webapp/channels/src/components/autosize_textarea.tsx

```diff
@@ -13,203 +14,169 @@ type Props = {
-            // Set an initial number of rows so that the textarea doesn't appear too large when its first rendered
-            heightProps.rows = 1;
-        } else {
-            heightProps.height = this.height;
-        }
+    if (height.current <= 0) {
+        // Set an initial number of rows so that the textarea doesn't appear too large when its first rendered
+        heightProps.rows = 1;
+    } else {
+        heightProps.height = height.current;
+    }
 
-        let textareaPlaceholder = null;
-        const placeholderAriaLabel = placeholder ? placeholder.toLowerCase() : '';
-        if (!this.props.value && !this.props.defaultValue) {
-            textareaPlaceholder = (
-                <div
-                    {...otherProps as any}
-                    id={`${id}_placeholder`}
-                    data-testid={`${id}_placeholder`}
-                    style={styles.placeholder}
-                >
-                    {placeholder}
-                </div>
-            );
-        }
+    let textareaPlaceholder = null;
+    const placeholderAriaLabel = placeholder ? placeholder.toLowerCase() : '';
+    if (!value && !defaultValue) {
+        textareaPlaceholder = (
+            <div
+                id={`${id}_placeholder`}
+                data-testid={`${id}_placeholder`}
+                style={styles.placeholder}
+            >
+                {placeholder}
+            </div>
+        );
+    }
 
-        return (
-            <div>
-                {textareaPlaceholder}
+    return (
+        <div>
+            {textareaPlaceholder}
+            <textarea
+                ref={setTextareaRef}
+                data-testid={id}
+                id={id}
+                {...heightProps}
+                role='textbox'
+                aria-label={placeholderAriaLabel}
+                dir='auto'
+                disabled={disabled}
+                onChange={onChange}
+                onInput={onInput}
+                value={value}
+                defaultValue={defaultValue}
+            />
+            <div style={styles.container}>
                 <textarea
-                    ref={this.setTextareaRef}
-                    data-testid={id}
-                    id={id}
-                    {...heightProps}
-                    {...otherProps}
-                    role='textbox'
-                    aria-label={placeholderAriaLabel}
+                    ref={referenceRef}
+                    id={id + '-reference'}
+                    style={styles.reference}
                     dir='auto'
-                    disabled={disabled}
-                    onChange={this.props.onChange}
-                    onInput={onInput}
-                    value={value}
-                    defaultValue={defaultValue}
+                    disabled={true}
+                    rows={1}
+                    value={value || defaultValue}
+                    aria-hidden={true}
+                    onChange={onChange}
```

Comment:
`onChange` passed here since it should come from `otherProps`, and this should keep any behaviour related to that.